tailnet/test/integration/integration.go

```diff
@@ -111,13 +111,43 @@ type SimpleServerOptions struct {
 
 var _ ServerStarter = SimpleServerOptions{}
 
+type connManager struct {
+	sync.Mutex
+	conns map[uuid.UUID]net.Conn
+}
+
+func (c *connManager) Add(id uuid.UUID, conn net.Conn) func() {
+	c.Lock()
+	c.conns[id] = conn
+	c.Unlock()
+	return func() {
+		c.Lock()
+		delete(c.conns, id)
+		c.Unlock()
+	}
+}
+
+func (c *connManager) Clear() {
+	c.Lock()
+	defer c.Unlock()
+	for _, conn := range c.conns {
+		_ = conn.Close()
+	}
+	c.conns = make(map[uuid.UUID]net.Conn)
+}
+
 //nolint:revive
 func (o SimpleServerOptions) Router(t *testing.T, logger slog.Logger) *chi.Mux {
 	coord := tailnet.NewCoordinator(logger)
 	var coordPtr atomic.Pointer[tailnet.Coordinator]
 	coordPtr.Store(&coord)
 	t.Cleanup(func() { _ = coord.Close() })
 
+	cm := connManager{
+		Mutex: sync.Mutex{},
```

Comment:
Since you're just using the zero value we usually don't bother initializing mutex fields in structs